src/Common/SipHash.h

```diff
@@ -161,15 +224,21 @@ class SipHash
         }
     }
 
-    template <typename T>
+    template <typename Transform = void, typename T>
     ALWAYS_INLINE void update(const T & x)
     {
         if constexpr (std::endian::native == std::endian::big)
         {
-            T rev_x = x;
-            char *start = reinterpret_cast<char *>(&rev_x);
-            char *end = start + sizeof(T);
-            std::reverse(start, end);
+            static constexpr auto instantiate_transform = std::invoke(
```

Comment:
If somehow possible, please no weird `std::invoke` (it's really rarely used throughout the codebase)